networkit/cpp/graph/Graph.cpp

```diff
@@ -1113,19 +1113,44 @@ void Graph::merge(const Graph &G) {
 
 // SUBGRAPHS
 
-Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes) const {
+Graph Graph::subgraphFromNodes(const std::unordered_set<node> &nodes, bool includeOutNeighbors, bool includeInNeighbors) const {
+	const auto neighbors = [&] {
+		std::unordered_set<node> neighbors;
+
+		if (!includeOutNeighbors && !includeInNeighbors)
+			return neighbors;
+
+		for (node u : nodes) {
+			if (includeOutNeighbors)
+				for(const node v : neighborRange(u))
+					neighbors.insert(v);
+
+			if (includeInNeighbors)
+				for(const node v : inNeighborRange(u))
+					neighbors.insert(v);
+		}
+
+		return neighbors;
+	}();
+
+	auto isRelevantNode = [&] (const node u) {
+		if (nodes.find(u) != nodes.end()) return 2;
+		if (!neighbors.empty() && neighbors.find(u) != neighbors.end()) return 1;
+		return 0;
+	};
```

Comment:
It might be worth to add a short comment explaining how the "relevance" of a node works.